src/core/storage/disk.test.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import {DiskStorage} from "./disk";
+import tmp from "tmp";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+import {tmpdir} from "os";
+// $FlowIgnore[missing-export]
+import {TextEncoder, TextDecoder} from "util";
+
+describe("core/storage/disk", () => {
+  function tmpWithContents(contents: mixed) {
+    const name = tmp.tmpNameSync();
+    fs.writeFileSync(name, JSON.stringify(contents));
+    return name;
+  }
+  const badPath = () => pathJoin(tmp.dirSync().name, "not-a-real-path");
+  const fooDefault = () => ({foo: 1337});
+  const getInstance = (baseDir?: string) => new DiskStorage(baseDir || "");
+  const basedir = tmp.dirSync();
+  const encoder = new TextEncoder();
+  // $FlowIgnore[incompatible-call]
+  const decoder = new TextDecoder();
+
+  it("DiskStorage accepts a basepath", () => {
+    const instance = new DiskStorage(tmpdir());
+    expect(instance._basePath).toBe(tmpdir());
+  });
+
+  describe("get", () => {
+    it("returns an existing file", async () => {
+      const instance = getInstance("");
+      const fname = tmpWithContents(fooDefault());
+
+      const result = await instance.get(fname);
+      expect(JSON.parse(decoder.decode(result))).toEqual(fooDefault());
+    });
+    it("throws if file doesn't exist", async () => {
+      const instance = getInstance();
+      const result = async () => await instance.get(badPath());
+      await expect(result).rejects.toThrow("ENOENT");
+    });
+  });
+  describe("set", () => {
```

Comment:
It’s okay to be using `TextEncoder`/`TextDecoder` for test data, but
this module really operates on bytes (as it must). Please include some
tests where the data is not valid JSON and not valid UTF-8. I want these
tests to catch an error given by, say, an accidental roundtrip through
encoding. If you don’t know where to start, take this building block:

```javascript
> var b1 = new Uint8Array([0x11]), b2 = new Uint8Array([0x99]);
> var roundtrip = (b) => new TextEncoder().encode(new TextDecoder().decode(b));
> [b1, roundtrip(b1)];
[ Uint8Array(1) [ 17 ], Uint8Array(1) [ 17 ] ]
> [b1, roundtrip(b2)];
[ Uint8Array(1) [ 17 ], Uint8Array(3) [ 239, 191, 189 ] ]
```

Reply:
@wchargin Thanks for this tip. I set one up as a sanity check, though I didn't throw an error. I just used it to demonstrate that some data that works with this implementation cannot be processed with Text encoder